tvtk/tvtk_base.py

```diff
@@ -184,13 +184,47 @@ class RevPrefixMap(PrefixMap):
     keys map to the same value, one of the valid keys will be used.
 
     """
-    def __init__(self, map, *extra_values, **kwargs):
-        super().__init__(map, **kwargs)
-        self._rmap = {}
-        for key, value in map.items():
-            self._rmap[value] = key
+    is_mapped = True
+
+    def __init__(self, map, *extra_values, default_value=None, **kwargs):
+        self.map = dict(map)
+        if not self.map:
+            raise ValueError(
+                "The dictionary of valid values can not be empty."
+            )
+
+        self._map = {key: key for key in self.map}
+        self._rmap = {value: key for key, value in self.map.items()}
+
+        if default_value is None:
+            default_value = next(iter(self.map))
+        else:
+            default_value = self.value_for(default_value)
+
         for key in extra_values:
-            self._rmap[key] = kwargs['default_value']
+            self._rmap[key] = default_value
+
+        super().__init__(default_value, **kwargs)
+
+    def value_for(self, value):
+        if not isinstance(value, str):
+            raise traits.TraitError(
+                "Value must be {}, but a value {!r} was specified.".format(
+                    self.info(), value)
+            )
+
+        if value in self._map:
+            return self._map[value]
+
+        matches = [key for key in self.map if key.startswith(value)]
+        if len(matches) == 1:
+            self._map[value] = match = matches[0]
```

Comment:
Traits 6.3.0 removed the caching from `PrefixMap`. We could do the same here. (But again, this would be a behaviour change that shouldn't be made in a bugfix release.)